server/src/test/java/org/opensearch/action/search/SearchQueryCategorizerTests.java

```diff
@@ -0,0 +1,205 @@
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        sourceBuilder.size(50);
+        BoostingQueryBuilder queryBuilder = new BoostingQueryBuilder(
+            new TermQueryBuilder("unmapped_field", "foo"),
+            new MatchNoneQueryBuilder()
+        );
+        sourceBuilder.query(queryBuilder);
+
+        searchQueryCategorizer.categorize(sourceBuilder);
+
+        Mockito.verify(searchQueryCategorizer.searchQueryCounters.otherQueryCounter, times(2)).add(eq(1.0d), any(Tags.class));
+        Mockito.verify(searchQueryCategorizer.searchQueryCounters.termCounter).add(eq(1.0d), any(Tags.class));
+    }
+
+    public void testQueryStringQuery() {
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        sourceBuilder.size(50);
+        QueryStringQueryBuilder queryBuilder = new QueryStringQueryBuilder("foo:*");
+        sourceBuilder.query(queryBuilder);
+
+        searchQueryCategorizer.categorize(sourceBuilder);
+
+        Mockito.verify(searchQueryCategorizer.searchQueryCounters.queryStringQueryCounter).add(eq(1.0d), any(Tags.class));
+    }
+
+    public void testRangeQuery() {
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        RangeQueryBuilder rangeQuery = new RangeQueryBuilder("date");
+        rangeQuery.gte("1970-01-01");
+        rangeQuery.lt("1982-01-01");
+        sourceBuilder.query(rangeQuery);
+
+        searchQueryCategorizer.categorize(sourceBuilder);
+
+        Mockito.verify(searchQueryCategorizer.searchQueryCounters.rangeCounter).add(eq(1.0d), any(Tags.class));
+    }
+
+    public void testRegexQuery() {
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        sourceBuilder.query(new RegexpQueryBuilder("field", "text"));
+
+        searchQueryCategorizer.categorize(sourceBuilder);
+
+        Mockito.verify(searchQueryCategorizer.searchQueryCounters.regexCounter).add(eq(1.0d), any(Tags.class));
+    }
+
+    public void testSortQuery() {
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        sourceBuilder.query(QueryBuilders.matchQuery("tags", "ruby"));
+        sourceBuilder.sort("creationDate", SortOrder.DESC);
+        sourceBuilder.sort(new ScoreSortBuilder());
+
+        searchQueryCategorizer.categorize(sourceBuilder);
+
+        Mockito.verify(searchQueryCategorizer.searchQueryCounters.matchCounter).add(eq(1.0d), any(Tags.class));
+        Mockito.verify(searchQueryCategorizer.searchQueryCounters.sortCounter).add(eq(1.0d));
+    }
+
+    public void testTermQuery() {
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        sourceBuilder.size(50);
+        sourceBuilder.query(QueryBuilders.termQuery("field", "value2"));
+
+        searchQueryCategorizer.categorize(sourceBuilder);
+
+        Mockito.verify(searchQueryCategorizer.searchQueryCounters.termCounter).add(eq(1.0d), any(Tags.class));
+    }
+
+    public void testWildcardQuery() {
```

Comment:
Thanks for adding the tests. Can we add another test case with multiple levels, for example: bool of term and match at level 2 and range at level 1 with aggregations?